src/modeler/AutoRest.Swagger/Validation/Core/Rule.cs

```diff
@@ -21,6 +22,18 @@ protected Rule()
         /// </summary>
         public virtual string Id => "!!! implement me and make me abstract !!!";
 
+
+        /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public virtual ServiceDefinitionDocumentType OpenApiDocumentValidationType => ServiceDefinitionDocumentType.Default;
+
+        /// <summary>
+        /// When to apply the validation rule, before or after it has been merged as a part of its composite document
```

Comment:
minor: I'd refrain myself from calling composite document as it confuses me with our legacy method. How aobut if we call "Merged" document from markdown file